daffodil-schematron/src/main/scala/org/apache/daffodil/validation/schematron/SchematronValidatorFactory.scala

```diff
@@ -75,7 +81,7 @@ object SchematronValidatorFactory {
     schematron: InputStream,
     schematronID: String,
     srcfmt: SchSource,
-    svrlPath: Option[Path]
+    svrlPath: Option[URI]
   ): SchematronValidator = {
     val factory = new TransformerFactoryImpl()
     factory.setURIResolver(DFDLCatalogResolver.get)
```

Comment:
This is an existing bug, but I *think* SchematronValidator is not thread safe. makeValidator is required to returns something that is thread safe. 

This creates a Schematron object which creates a Transformer. And the javadoc says transformer is not thread safe. 

So I think the transformer needs to be wrapped in a ThreadLocal or something. 

It it does wrap the XMLReader in a ThreadLocal, but it only gets an instance from that ThreadLocal when it initially creates the Schematron object, which is shared among thread. So I don't thin kthe use of the XMLReader is works probably.

Can you verify that I'm actually reading this all correctly, and if so, open a bug to fix this?

Reply:
DAFFODIL-3009